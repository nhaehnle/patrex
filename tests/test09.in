
p = new(sizeof(Foo), mem0) Foo(arg0, arg1);

q = new(sizeof(*q), mem1) Sub::Holder<Bar>(arg0);
