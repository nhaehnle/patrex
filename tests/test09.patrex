define arg $!(,)*
define notlist $!( ( $.* ) )
define notlist_angle $!( $|( ( $.* ) )(>) )

match $( new ( sizeof($.*), ${arg}|mem|) $( ${notlist}* ${notlist_angle} )|type| )|all| ($.*)
	replace all "make<{type}>({mem})"

match $( new ( sizeof($.*), ${arg}|mem|) $( ${notlist}* > )|type| )|all| ($.*)
	replace all "make<{type} >({mem})"
