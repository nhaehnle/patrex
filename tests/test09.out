
p = make<Foo>(mem0)(arg0, arg1);

q = make<Sub::Holder<Bar> >(mem1)(arg0);
